core/src/main/java/org/axonframework/commandhandling/model/inspection/MethodCommandHandlerDefinition.java

```diff
@@ -61,7 +63,8 @@ private MethodCommandMessageHandlingMember(MessageHandlingMember<T> delegate,
             } else {
                 commandName = (String) annotationAttributes.get("commandName");
             }
-            isFactoryHandler = (executable instanceof Constructor);
+            final boolean factoryMethod = executable instanceof Method && Modifier.isStatic(executable.getModifiers());
```

Comment:
I'm wondering if checking for the `static` modifier is enough. Should we check for compatibility of the return type with the type of Aggregate Root that declares is?
I'm inclined to say that having a return type that is not assignable to the type of class declaring it, is a configuration error.